generators/server/templates/src/main/resources/config/_application.yml

```diff
@@ -186,19 +189,31 @@ security:
     oauth2:
         resource:
             filter-order: 3
-    <%_ } else if (authenticationType === 'oauth2') { _%>
+    <%_ } _%>
+    <%_ if (authenticationType === 'oauth2') {  _%>
     oauth2:
         client:
             accessTokenUri: ${oauth2.issuer}/protocol/openid-connect/token
             userAuthorizationUri: ${oauth2.issuer}/protocol/openid-connect/auth
+            <%_ if (applicationType === 'gateway') { _%>
```

Comment:
`clientId`, `clientSecret`, `clientAuthenticationScheme`, and `scope` should also be included for a monolith, but `tokenName` is not needed.